mypy/semanal.py

```diff
@@ -1094,14 +1095,16 @@ def visit_decorator(self, dec: Decorator) -> None:
                     removed.append(i)
                 else:
                     self.fail("@final cannot be used with non-method functions", d)
+            elif not dec.var.is_property:
```

Comment:
if a decorator shows up before "is_property" gets set then it's

@decorator
@property
def foo(...)

else we are in the propertied decorator scenario:

@property
@decorator
def foo(...)